src/services/posts/fetch-posts.js

```diff
@@ -0,0 +1,44 @@
+const db = require("../../db")
+
+const fetchPosts = async(isFlagged) => {
+	
+	let feed 
+	if(isFlagged){
+		feed = await db.query(`
+		SELECT posts.id, posts."userId" , posts.title, posts.image,
+		posts.content,posts.published, posts."createdAt", posts.type,
+		posts."likesCount", posts."flagsCount",
+		users."firstName", users."lastName", users.email,
+		users."profilePictureUrl" 
+		FROM posts
+		INNER JOIN users on posts."userId" = users.id
+		WHERE posts."flagsCount" > 0
+		ORDER BY "flagsCount" DESC;
+		`)
+	}
+	else if(isFlagged === false){
+		feed = await db.query(`
+		SELECT posts.id, posts."userId" , posts.title, posts.image,
+		posts.content,posts.published, posts."createdAt", posts.type,
+		posts."likesCount", posts."flagsCount",
+		users."firstName", users."lastName", users.email,
+		users."profilePictureUrl" 
+		FROM posts
+		INNER JOIN users on posts."userId" = users.id
+		WHERE posts."flagsCount" = 0
+		ORDER BY "flagsCount" DESC;
+		`)
+	}
+	else(
+		feed = await db.query(`
+	SELECT posts.id, posts."userId" , posts.title, posts.image, posts.content,
+	posts.published, posts."createdAt", posts.type, posts."likesCount",
+	users."firstName", users."lastName", users.email, users."profilePictureUrl" 
+	FROM posts
+	INNER JOIN users on posts."userId" = users.id;
+	`)
```

Comment:
`isFlagged !== undefined`

I have not seen this kind of javascript expression before. Is it the same as `isFlagged && isFlagged !== undefined` ?

Reply:
The example written is amazing, I keep admiring the code :sweat_smile: :sweat_smile: :sweat_smile:

Reply:
> `isFlagged !== undefined`
> 
> I have not seen this kind of javascript expression before.

it is actually just concerned about checking if the `isFlagged` parameter is set, no matter the value, whether false or true or anything. If it is not set, it's value will be `undefined`. Don't know if you've seem checks like `if(typeof variable === 'undefined')`

so if we check and it is not defined (!==undefined), means it was set to a value. 

>  Is it the same as `isFlagged && isFlagged !== undefined` ?

yea not in every case, when you have a boolean being false, the first `isFlagged` part will fail and the whole condition will not run. 

Its a common oversight when it comes to booleans because mostly we check almost every variable to see if its defined by using the condition like `if(isFlagged)` but when its a boolean and it is intended for both `true` or `false`, it will only run if the value is `true`

Reply:
oh woow! that makes perfect sense when it comes to dealing with boolean and I see how it is a common oversight. Thanks for the explanation.  I get it now